go-controller/pkg/ovn/gress_policy.go

```diff
@@ -58,14 +62,63 @@ func newGressPolicy(policyType knet.PolicyType, idx int, namespace, name string)
 		policyName:            name,
 		policyType:            policyType,
 		idx:                   idx,
-		peerAddressSets:       make(map[string]bool),
+		nsAddressSets:         sets.String{},
 		sortedPeerAddressSets: make([]string, 0),
 		portPolicies:          make([]*portPolicy, 0),
 		ipBlockCidr:           make([]string, 0),
 		ipBlockExcept:         make([]string, 0),
 	}
 }
 
+func (gp *gressPolicy) ensurePeerAddressSet(factory AddressSetFactory) error {
+	if gp.peerAddressSet != nil {
+		return nil
+	}
+
+	direction := strings.ToLower(string(gp.policyType))
+	asName := fmt.Sprintf("%s.%s.%s.%d", gp.policyNamespace, gp.policyName, direction, gp.idx)
+	as, err := factory.NewAddressSet(asName, nil)
+	if err != nil {
+		return err
+	}
+
+	gp.peerAddressSet = as
+	gp.sortedPeerAddressSets = append(gp.sortedPeerAddressSets, as.GetHashName())
```

Comment:
@trozet probably yes, but I'd rather do that after we decouple the networkpolicy.

It's OK right now because ensurePeerAddressSet() is only called from the NetworkPolicy object add handler before any other things have access to the gressPolicy. The only other accesses are done under the NamespacePolicy lock and will not conflict.

Reply:
I think if you are refactoring this to be address set objects, with a manager "factory" to handle tracking all of the objects with locks, we should access address sets with proper locking. Maybe I'm misunderstanding the intention of the factory and holding the sets in it, that's why I was waiting for your responses to these comments before I reviewed further.